File: apps/ios/Burcum/Models/HoroscopeReading.swift
import Foundation

struct HoroscopeReading: Codable, Identifiable {
    let id: String
    let zodiacSign: String
    let readingType: ReadingType
    let content: String
    let advice: String?
    let loveScore: Int?
    let careerScore: Int?
    let healthScore: Int?
    let luckyNumbers: [Int]?
    let luckyColor: String?
    let createdAt: String
}

enum ReadingType: String, Codable {
    case daily
    case weekly
    case monthly
    case yearly
}

struct APIResponse<T: Codable>: Codable {
    let success: Bool
    let data: T?
    let error: String?
    let cached: Bool?
}

struct CompatibilityResult: Codable {
    let sign1: String
    let sign2: String
    let overallScore: Int
    let loveScore: Int
    let friendshipScore: Int
    let workScore: Int
    let analysis: String
    let strengths: [String]
    let challenges: [String]
    let advice: String
}

File: apps/android/app/src/main/java/com/burcum/app/model/HoroscopeReading.kt
package com.burcum.app.model

import kotlinx.serialization.Serializable

@Serializable
data class HoroscopeReading(
    val id: String,
    val zodiacSign: String,
    val readingType: String,
    val content: String,
    val advice: String? = null,
    val loveScore: Int? = null,
    val careerScore: Int? = null,
    val healthScore: Int? = null,
    val luckyNumbers: List<Int>? = null,
    val luckyColor: String? = null,
    val createdAt: String
)

@Serializable
data class ApiResponse<T>(
    val success: Boolean,
    val data: T? = null,
    val error: String? = null,
    val cached: Boolean? = null
)

@Serializable
data class CompatibilityResult(
    val sign1: String,
    val sign2: String,
    val overallScore: Int,
    val loveScore: Int,
    val friendshipScore: Int,
    val workScore: Int,
    val analysis: String,
    val strengths: List<String>,
    val challenges: List<String>,
    val advice: String
)

@Serializable
data class User(
    val id: String,
    val email: String,
    val name: String? = null,
    val zodiacSign: String? = null,
    val birthDate: String? = null,
    val subscriptionTier: SubscriptionTier = SubscriptionTier.FREE,
    val emailVerified: Boolean = false
)

@Serializable
enum class SubscriptionTier {
    FREE, PREMIUM, VIP
}

@Serializable
data class LoginRequest(
    val email: String,
    val password: String
)

@Serializable
data class RegisterRequest(
    val email: String,
    val password: String,
    val name: String? = null,
    val birthDate: String? = null
)

@Serializable
data class AuthResponse(
    val user: User,
    val token: String
)

@Serializable
data class CompatibilityRequest(
    val sign1: String,
    val sign2: String
)
